python/ray/autoscaler/batching_node_provider.py

```diff
@@ -160,6 +164,13 @@ def non_terminated_nodes(self, tag_filters: Dict[str, str]) -> List[str]:
             workers_to_delete=set(),  # No workers to delete yet
         )
         all_nodes = list(self.node_data_dict.keys())
+        # Initialize map of replica indices to nodes in that replica
+        self.replicas_to_nodes = defaultdict(list)
```

Comment:
How about initializing `replicas_to_nodes` in the constructor?

Reply:
Done.